test/matchers/to-have-status-effect-matcher.ts

```diff
@@ -0,0 +1,53 @@
+// -- start tsdoc imports --
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Pokemon } from "#app/field/pokemon";
+// -- end tsdoc imports --
+
+import { StatusEffect } from "#enums/status-effect";
+import { isPokemonInstance, receivedStr } from "#test/test-utils/testUtils";
+import type { SyncExpectationResult } from "@vitest/expect";
+
+//#region Types
+
+export interface ToHaveStatusEffectMatcherOptions {
+  ignoreMockAbility?: boolean;
+}
+
+//#endregion
+//#region Exports
+
+/**
+ * Matcher to check if a Pokemon's {@linkcode StatusEffect} is as expected
+ * @param received - The actual value received. This matcher expects this
+ * value to be a {@linkcode Pokemon}
+ * @param expectedStatusEffect - The {@linkcode StatusEffect} the Pokemon is
+ * expected to have
+ * @param ignoreMockAbility - If `true`, ignores the effects of abilities that
+ * mock status effects (i.e. Comatose)
+ * @returns the results of the matcher's assertion
+ */
+export function toHaveStatusEffectMatcher(
+  received: unknown,
```

Comment:
Add `this: MatcherState` to have type-hints for `this`. `expect.extend()` invokes it anyway

```suggestion
  this: MatcherState,
  received: unknown,
```